src/renderer/containers/AddBeaconNode/AddBeaconNode.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useState} from "react";
+import {useDispatch} from "react-redux";
+import {useHistory, useParams} from "react-router";
+import {saveBeaconNodeAction, startBeaconChainAction} from "../../actions/network";
+import {ConfigureBeaconNode} from "../../components/ConfigureBeaconNode/ConfigureBeaconNode";
+import {InputBeaconNode} from "../../components/ConfigureBeaconNode/InputBeaconNode";
+import {Routes} from "../../constants/routes";
+import {Container} from "../../services/docker/container";
+import {DockerPort} from "../../services/docker/type";
+import OnBoardModal from "../Onboard/OnBoardModal";
+
+export const AddBeaconNodeContainer: React.FunctionComponent = () => {
+    const {validatorKey} = useParams();
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const [currentStep, setCurrentStep] = useState<number>(0);
+    const [network, setNetwork] = useState<string|undefined>();
+
+    const renderFirstStep = (): React.ReactElement => {
+        const onRunNodeSubmit = async(network: string): Promise<void> => {
+            setNetwork(network);
+
+            if (await Container.isDockerInstalled()) {
+                setCurrentStep(1);
+            } else {
+                // TODO: Configure Docker path?
+            }
+        };
+
+
+        const onGoSubmit = async(beaconNodeInput: string): Promise<void> => {
+            dispatch(saveBeaconNodeAction(beaconNodeInput, undefined, validatorKey));
+            history.push(Routes.DASHBOARD_ROUTE);
+        };
+
+        return (
+            <InputBeaconNode
+                onGoSubmit={onGoSubmit}
+                onRunNodeSubmit={onRunNodeSubmit}
+            />
+        );
+    };
+
+    const renderSecondStep = (): React.ReactElement => {
+        const onDockerRunSubmit = (ports: DockerPort[], libp2pPort: string, rpcPort: string): void => {
```

Comment:
isn't this suppose to be useCallback so it get's recreated if history changes?

Reply:
Yeah, that would be optimization